test/distributed/_composable/fully_shard/test_fully_shard_compile.py

```diff
@@ -39,9 +39,11 @@ def test_compile(self):
                     ShardingStrategy.FULL_SHARD,
                     ShardingStrategy.SHARD_GRAD_OP,
                     ShardingStrategy.NO_SHARD,
+                    ShardingStrategy.HYBRID_SHARD,
+                    ShardingStrategy._HYBRID_SHARD_ZERO2,
```

Comment:
How do these get tested in CI or even with a single node locally? What process groups get constructed?

Is the replication group a singleton process group and the sharding process group takes all GPUs?

Reply:
(I think the answer to this affects whether we should land this PR as is, or if we need to do some changes on manually specifying PGs.)

Reply:
that's a good point, I will check this. I ran it on a single node so I wonder if it just created an FSDP group and singel rank data parallel groups.

Reply:
@awgu I think we just need a similar world size change as in the prev diff.